### PR TITLE
Add GPT-4o as possible model

### LINKED ISSUE #11766: Add support for `openai/gpt-4o` model on the assistant dropdown
### Check for existing issues

- [X] Completed

### Describe the feature

Add support for the new `openai/gpt-4o` (alternatively, let us select any model name).

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by nudded**
Resolves: #11766

Release Notes:

- Add GPT-4o support (see: https://openai.com/index/hello-gpt-4o/). GPT-4o is better and faster than 4-turbo, at half the price.

### CODE REVIEW

**Comment by nudded** [comment]
@cla-bot check

File: assets/settings/default.json [hunk 1]
```diff
@@ -300,7 +300,8 @@
       // 1. "gpt-3.5-turbo"
       // 2. "gpt-4"
       // 3. "gpt-4-turbo-preview"
-      "default_model": "gpt-4-turbo-preview"
+      // 4. "gpt-4o"
+      "default_model": "gpt-4o"
```
Comment (mrnugget) [inline assets/settings/default.json:304]: Should we really change the default model?

Comment (nudded) [reply assets/settings/default.json:304]: Yeah, I've debated this myself. My thinking was, it's equally capable as 4-turbo, but faster and cheaper. Feels like a nice win for people just using the default config.

Comment (mrnugget) [reply assets/settings/default.json:304]: Makes sense, yeah.

**Pull request merged**